src/error.rs

```diff
@@ -59,6 +59,10 @@ pub enum Error {
     /// never be done in production
     DatabaseNotFound,
 
+    /// Returned if attempting to access a database driver for a different database type
+    /// than the one configured.
+    DatabaseMismatch,
```

Comment:
I think it may be possible, and definitely beneficial if we can, to combine this with DatabaseNotFound, rather than adding a new Error variant.  If we really do need to add an error variant, it should follow the same NounVerb naming convention as the others, not NounNoun.

Reply:
Agreed.

Reply:
Fixed.